client/components/LearnerViewDirective.js

```diff
@@ -410,11 +469,21 @@ tie.directive('learnerView', [function() {
           loadQuestion(questionId, questionSet.getIntroductionParagraphs());
         };
 
-        $scope.submitCode = function(code) {
+        var callSolutionHandlerService = function() {
+          var code = this.code;
           SolutionHandlerService.processSolutionAsync(
-            prompts[currentPromptIndex], code,
-            question.getAuxiliaryCode(language), language
-          ).then(setFeedback);
+              prompts[currentPromptIndex], code,
+              question.getAuxiliaryCode(language), language
+              ).then(setFeedback);
+        };
+
+        $scope.submitCode = function(code) {
+          $scope.hideDotDiv = false;
+          $timeout(
+              function(){feedbackDiv.scrollTop = feedbackDiv.scrollHeight + 17},
+              0);
+          var serviceCall = callSolutionHandlerService.bind({'code': code});
+          $timeout(serviceCall, 1000);
```

Comment:
Oh, I see, that's odd, but I think it's because some DOM work is going on here.

I think 15-20 is fine -- that's much less noticeable than 1000ms. You could perhaps do something like this:

```
        $scope.submitCode = function(code) {
          $scope.hideDotDiv = false;
          $timeout(function() {
            feedbackDiv.scrollTop = feedbackDiv.scrollHeight + 17;
            $timeout(function() {
              SolutionHandlerService.processSolutionAsync(
                prompts[currentPromptIndex], code,
                question.getAuxiliaryCode(language), language
              ).then(setFeedback);
            }, 10);
          }, 0);
        };
```